includes/collection/class-inbox.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Inbox collection file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Collection;
+
+use Activitypub\Webfinger;
+use Activitypub\Activity\Activity;
+use Activitypub\Activity\Base_Object;
+
+use function Activitypub\object_to_uri;
+use function Activitypub\is_activity_public;
+
+/**
+ * ActivityPub Inbox Collection
+ *
+ * @link https://www.w3.org/TR/activitypub/#inbox
+ */
+class Inbox {
+	const POST_TYPE = 'ap_inbox';
+
+	/**
+	 * Add an activity to the inbox.
+	 *
+	 * @param Activity|\WP_Error $activity The Activity object.
+	 * @param int                $user_id  The id of the local blog-user.
+	 *
+	 * @return false|int|\WP_Error The added item or an error.
+	 */
+	public static function add( $activity, $user_id ) {
+		if ( \is_wp_error( $activity ) ) {
+			return $activity;
+		}
+
+		// Check for duplicate activity.
+		if ( ! \is_wp_error( self::get_by_guid( $activity->get_id() ) ) ) {
+			return false;
```

Comment:
If there's already a post, should we return that post_id here? Or a WP_Error? Would be nice to not have to account for that `false`.